addons/website_sale/static/src/js/extra_info.js

```diff
@@ -1,76 +1,45 @@
-import publicWidget from "@web/legacy/js/public/public_widget";
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
 
-publicWidget.registry.websiteSaleExtraInfo = publicWidget.Widget.extend({
-    // /shop/extra_info
-    selector: '#class_extra_info',
 
-    /**
-     * @constructor
-     */
-    init: function () {
-        this._super.apply(this, arguments);
-
-        this.http = this.bindService('http');
-
-        this.submitButton =  document.getElementsByName('website_sale_main_button')[0]
-        this._boundSaveExtraInfo = this._onSaveExtraInfo.bind(this);
-        this.submitButton.addEventListener("click", this._boundSaveExtraInfo);
-        this.extraForm = document.querySelector('form.form_extra_info');
-    },
-
-    destroy() {
-        this.submitButton.removeEventListener("click", this._boundSaveExtraInfo);
-        this._super(...arguments);
-    },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
+export class WebsiteSaleExtraInfo extends Interaction {
```

Comment:
I wonder if we shouldn't extends `Form` from `website` directly 🤔

Reply:
Having issues when extending Form. Uncaught Promise > this.el.reset is not a function.
Maybe it's more reasonable to extend from Interaction :eyes: than freshly refactored (not tested?) files? :see_no_evil: